components/layout/inline.rs

```diff
@@ -687,18 +687,23 @@ impl LineBreaker {
         let mut need_ellipsis = false;
         let available_inline_size = self.pending_line.green_zone.inline -
             self.pending_line.bounds.size.inline - indentation;
-        match (fragment.style().get_text().text_overflow,
-               fragment.style().get_box().overflow_x) {
-            (text_overflow::T::clip, _) | (_, overflow_x::T::visible) => {}
-            (text_overflow::T::ellipsis, _) => {
+
+        let text_overflow_string: Option<String> = match fragment.style().get_text().text_overflow.first {
+            longhands::text_overflow::Side::Clip => None,
+            longhands::text_overflow::Side::Ellipsis => {
```

Comment:
can we write it like:

```rust
let need_ellipsis = match fragment.style().get_text().text_overflow.first {
    longhands::text_overflow::Side::Clip => false,
    longhands::text_overflow::Side::Ellipsis | longhands::text_overflow::Side::String(..) => ragment.margin_box_inline_size() > available_inline_size,
};
```

And then stop passing the string to `transform_into_ellipsis`, and look into `self.style()` there?

Also, this stops checking for `overflow-x: visible`, which I think is wrong, and would need a spec link if not.